crates/core/src/operations/optimize.rs

```diff
@@ -321,26 +338,20 @@ impl<'a> std::future::IntoFuture for OptimizeBuilder<'a> {
             let operation_id = this.get_operation_id();
             this.pre_execute(operation_id).await?;
 
-            let writer_properties = this.writer_properties.unwrap_or_else(|| {
-                WriterProperties::builder()
-                    .set_compression(Compression::ZSTD(ZstdLevel::try_new(4).unwrap()))
```

Comment:
For completeness, we probably want a way for FileFormatOptions / WriterPropertiesFactory to use different compression or other properties for files written during compaction.

I guess users can just modify the options passed when the set up the `DeltaOps`, so maybe that's enough and we don't need to change the API to support this when reusing the same options for different ops.